packages/iris-grid/src/IrisGrid.tsx

```diff
@@ -2336,7 +2335,9 @@ export class IrisGrid extends Component<IrisGridProps, IrisGridState> {
     if (partitionColumn == null) {
       return;
     }
-    onPartitionAppend(partitionColumn, value);
+    if (onPartitionAppend) {
+      onPartitionAppend(partitionColumn, value);
+    }
```

Comment:
You can just write this with optional chaining:
```suggestion
    onPartitionAppend?.(partitionColumn, value);
```